go.mod

```diff
@@ -22,8 +24,9 @@ require (
 	k8s.io/client-go v0.31.1
 	k8s.io/code-generator v0.31.1
 	k8s.io/gengo v0.0.0-20240826214909-a7b603a56eb7
-	k8s.io/gengo/v2 v2.0.0-20240228010128-51d4e06bde70
+	k8s.io/gengo/v2 v2.0.0-20240826214909-a7b603a56eb7
 	k8s.io/kube-aggregator v0.31.1
+	k8s.io/kube-openapi v0.0.0-20240903163716-9e1beecbcb38
```

Comment:
I'd suggest keeping the version closer to our targeted k8s.io dependency. For example for v0.31.1, they're using [k8s.io/kube-openapi v0.0.0-20240228011516-70dd3763d340](https://github.com/kubernetes/kubernetes/blob/v1.31.1/go.mod#L109). Unless this prevent the PR from working.

Reply:
Yeah, this makes the PR to not to work, looks like it's something to gengo version.

Reply:
Can you elaborate? Downgrading the kube-openapi version to that one seems to work fine for me (while keeping gengo with the version you have)